File: C#/01-Classes/Estatico.cs
using System;

//O uso do static em um método indica que não é necessario instanciar
//um objeto da classe a qual ele pertence utiliza-lo.
class Carro 
{
    public string marca;
    public string modelo;

    //Este método é estático
    public static void desligado() 
    {
        Console.WriteLine("Motor desligado");
    }

    //Este não
    public string ligarMotor() 
    {
        return "Motor ligado";
    }
}

class Programa
{

    static void Main(string[] args)
    {
        //Método estático já pode ser invocado. Não esquecer de colocar a classe a qual ele pertence
        Carro.desligado();

        //Agora o outro método só pode ser invocado se instanciarmos um objeto primeiro
        Carro carro1 = new Carro();
        carro1.marca = "Mazda";
        carro1.modelo = "RX-8";
        Console.WriteLine("{0} {1} : {2}", carro1.marca, carro1.modelo, carro1.ligarMotor());
        Console.ReadLine();
    }
}

File: Java/01-Classes/Estatico.java
class Caneta {
    String modelo;
    String cor;

    //O uso de static indica que o método é estatico, não é necessario
    //instanciar um objeto desta classe para usar este método
    static void metodoEstatico() {
        System.out.println("Este metodo e estatico");
    }

    //Este não é portanto so pode ser chamado depois de instanciar um objeto
    void escrever() {
        System.out.println("Escrevendo...");
    }
}

class Estatico {
    public static void main(String[] args) {
        //Este método é estatico e ja pode ser chamado
        Caneta.metodoEstatico();

        //Este outro só pode ser chamado se instanciarmos um objeto primeiro
        Caneta c1 = new Caneta();
        c1.modelo = "BIC";
        c1.cor = "preta";
        c1.escrever();
    }
}